src/Compilers/Core/Portable/SourceGeneration/IncrementalStepRunReason.cs

```diff
@@ -10,7 +10,7 @@ namespace Microsoft.CodeAnalysis
     public enum IncrementalStepRunReason
     {
         /// <summary>
-        /// The output of this step is a new output produced from a new input.
+        /// The input to this step was added or modified from a previous run, and it produced a new output.
```

Comment:
This is still not very clear. There's some ambiguity between an input node and the entries that it output.
@chsienki Would it make sense to simplify (here and below) to: "The input to this step produced a new output."

Reply:
I tried to follow the same format as the other enum values here. E.g. "The input to this step was modified from a previous run" is used twice below, for `Modified` and `Unchanged`. I personally find the symmetry between the documentation strings useful for understanding them.

I don't really get what could be considered ambiguous here, but  it could be that I'm just misunderstanding something.